src/Npgsql/Schema/DbColumnSchemaGenerator.cs

```diff
@@ -170,38 +178,37 @@ internal ReadOnlyCollection<NpgsqlDbColumn> GetColumnSchema()
             return result.AsReadOnly()!;
         }
 
-        NpgsqlDbColumn LoadColumnDefinition(NpgsqlDataReader reader, NpgsqlDatabaseInfo databaseInfo, bool oldQueryMode)
+        async ValueTask<NpgsqlDbColumn> LoadColumnDefinition(NpgsqlDataReader reader, NpgsqlDatabaseInfo databaseInfo, bool oldQueryMode, bool async, CancellationToken cancellationToken = default)
```

Comment:
As above, no need for all these to be async calls, and so this method can be synchronous.